ADR/0022-secret-mgmt-for-user-workloads.md

```diff
@@ -26,65 +29,142 @@ Accepted
 #### Architecture
 
 ![](../diagrams/secret-mgmt.excalidraw.svg)
+TODO update diagram
 
-The idea is to introduce a new CR called `RemoteSecret`, which is a Kubernetes representation of the `K8s Secret` that is stored in permanent storage aka SecretStorage. This CR contains a reference to the destination, either a Kubernetes namespace or an `Environment`. A temporary **Upload Secret** is used to perform an upload to the permanent storage and is represented as a regular Kubernetes secret with special labels and annotations recognized by the _SPI controller_. Different implementations of `SecretStorage` can be used, such as AWS Secret Manager or HashiCorp Vault.
+The proposed solution is to create a new Kubernetes Custom Resource (CR) called `RemoteSecret`. It serves as a representation of the Kubernetes Secret that is stored in permanent storage, which is also referred to as `SecretStorage`. This Custom Resource includes references to targets, like Kubernetes namespaces, that may also contain the required data to connect to a remote Kubernetes. To perform an upload to permanent storage, a temporary **Upload Secret** is utilized, which is represented as a regular Kubernetes Secret with special labels and annotations that the SPI controller recognizes. Different `SecretStorage` implementations, like AWS Secret Manager or HashiCorp Vault, can be used. It is simpler to create the RemoteSecret first and then use the linked **Upload Secret** to upload secret data. However, in simple cases, the **Upload Secret**  can be used to perform both uploading and the creation of RemoteSecret in a single action. It's worth noting that the **Upload Secret** is not a core component of the framework, but rather a convenient way of creating secrets. In the future, it is possible that new methods of uploading SecretData to RemoteSecret may be added.
 
-#### Example: If the destination is `Environment`
+
+
+#### Example: If the working Secret should be created along with RemoteSecret.
 ```yaml
-apiVersion: v1
-kind: Secret
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
 metadata:
-  name: tmp-jdbc-connection-parameter-upload
-  labels:
-    spi.appstudio.redhat.com/upload-secret: secret
-    spi.appstudio.redhat.com/target-environment: prod
-    spi.appstudio.redhat.com/target-secret-name: jdbc-connection-parameter
-type: Opaque
-data:
- ...
+    namespace: jdoe-tenant
+spec:
+    target:
+        - namespace: spi-test-target
+          secretName: test-remote-secret-secret-5bmq9
+status:
+    TBD
 ```
+> :warning: **Note for this and following:** `RemoteSecret` is not Ready, `SecretData` referenced to it has to be uploaded. See `Uploading secret data to RemoteSecret` example.
```

Comment:
maybe explain somewhere what `Ready` means for the remote secret.

Reply:
I think there are 2 means of Status:
- related to StoredData. Ready would mean it is stored (ready to use by RemoteSecret)
- related to Targets. Ready would mean (per each) the Secret is created (ready to use by business logic)